### PR TITLE
[NVPTX] Promote v2i8 to v2i16

### PR DESCRIPTION
**Pull request by manasij7479**
Promote v2i8 to v2i16, fixes a crash. 
Re-enable a test in NVPTX/vector-returns.ll

https://github.com/llvm/llvm-project/issues/104864

### CODE REVIEW

**Review by justinfargnoli — APPROVED** [review]
Please wait for @Artem-B to review before merging.

File: llvm/test/CodeGen/NVPTX/vector-returns.ll
Comment (Artem-B) [inline llvm/test/CodeGen/NVPTX/vector-returns.ll:337]: It may be useful to use a non-zero value, so we can verify that we've stored the vector elements in correct order.
That, actually applies to all the tests here, so it may be worth fixing that in a separate patch.

**Review by Artem-B — APPROVED** [review]

**Pull request merged**
